spring-integration-core/src/main/java/org/springframework/integration/endpoint/IntegrationConsumer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.endpoint;
+
+import org.springframework.integration.handler.AbstractMessageHandler;
+
+/**
+ * Consumer.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+public interface IntegrationConsumer {
```

Comment:
Nice! Ar a glance I didn't liked this name, because you use it to build a *runtime model*, but now I see that is just like a side effect. And this contract really represents the `Consumer` like it is the SI.

Although its content looks a bit ugly still just with these getters...
But we don't have choice and there is no more (yet) functionality to expose.

So, this one LGTM. :smile: